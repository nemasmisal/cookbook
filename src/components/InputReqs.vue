<template>
  <div class="activator">
    <button @click="toggleReveal">
      Need help?
      <i class="material-icons">mode_comment</i>
    </button>
  </div>
  <transition name="drop">
    <div class="info" v-if="reveal">
      <template v-for="field in fields" :key="field">
        <h2>{{ field.name }}</h2>
        <ol>
          <li v-for="req in field.reqs" :key="req">{{ req }}</li>
        </ol>
      </template>
    </div>
  </transition>
</template>
<script>
import { ref } from 'vue';
export default {
  props: { fields: { type: Object } },
  setup() {
    const reveal = ref(false);
    const toggleReveal = () => {
      reveal.value = !reveal.value;
    };
    return { reveal, toggleReveal };
  },
};
</script>
<style lang="stylus" scoped>
.drop-enter-from
  transform translateY(-100%)
.drop-enter-to
  transform translateY(0%)
.drop-enter-active
  transition all 1s ease
.drop-leave-from
  transform translateY(0%)
.drop-leave-to
  transform translateY(-100%)
.drop-leave-active
  transition all 1s ease
.info
  position relative
  top 0
  background #f8f8f8
  padding 10px
  z-index -1
h2,h3
  margin 5px
  padding 5px
li
  list-style circle
  margin 0 auto
  max-width 300px
  text-align start
  font-size 15px
.activator
  background #f8f8f8
  width 94%
  display flex
  justify-content flex-end
  padding-top 20px
  padding-right 20px
  align-items center
  > button
      background none
      margin 0
      border none
      color #2c3e50
      padding 0
      &:hover
        cursor pointer
</style>
