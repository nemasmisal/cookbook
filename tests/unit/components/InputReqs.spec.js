import { shallowMount } from '@vue/test-utils';
import InputReqs from '@/components/InputReqs.vue';

describe('E2E testing', () => {
  let fields;
  let wrapper;
  beforeEach(() => {
    fields = [{ name: 'test name', reqs: ['first req', 'second req'] }];
    wrapper = shallowMount(InputReqs, {
      props: { fields },
    });
  });
  describe('Testing component props', () => {
    it('Should accept/use passed props', () => {
      expect(wrapper.vm.fields).toMatchObject(fields);
    });
    it('Should have inital reveal value to false', () => {
      expect(wrapper.vm.reveal).toBeFalsy();
    });
  });
  describe('Testing component methods', () => {
    it('Should set reveal value to its oposite', async () => {
      expect(wrapper.vm.reveal).toBeFalsy();
      await wrapper.vm.toggleReveal();
      expect(wrapper.vm.reveal).toBeTruthy();
      await wrapper.vm.toggleReveal();
      expect(wrapper.vm.reveal).toBeFalsy();
    });
  });
  describe('Testing component functionallity from DOM event', () => {
    it('Should render reveal after click then hide it after another', async () => {
      await wrapper.find('button').trigger('click');
      expect(wrapper.element).toMatchSnapshot();
      await wrapper.find('button').trigger('click');
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
