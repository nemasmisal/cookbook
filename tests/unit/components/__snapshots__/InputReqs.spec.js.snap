// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E testing Testing component functionallity from DOM event Should render reveal after click then hide it after another 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="activator"
  >
    <button>
       Need help? 
      <i
        class="material-icons"
      >
        mode_comment
      </i>
    </button>
  </div>
  <transition-stub>
    <div
      class="info"
    >
      
      
      <h2>
        test name
      </h2>
      <ol>
        
        <li>
          first req
        </li>
        <li>
          second req
        </li>
        
      </ol>
      
      
    </div>
  </transition-stub>
  
</div>
`;

exports[`E2E testing Testing component functionallity from DOM event Should render reveal after click then hide it after another 2`] = `
<div
  data-v-app=""
>
  
  <div
    class="activator"
  >
    <button>
       Need help? 
      <i
        class="material-icons"
      >
        mode_comment
      </i>
    </button>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  
</div>
`;
